assets/model_monitoring/components/src/model_monitor_metric_outputter/builder/metric_output_builder.py

```diff
@@ -35,34 +35,31 @@ def _build(self, metrics: List[Row]) -> dict:
         metrics_dict = {}
         for metric in metrics:
             metric_dict = metric.asDict()
-            metric_name = metric_dict[SIGNAL_METRICS_METRIC_NAME]
 
-            group_names = []
-            if SIGNAL_METRICS_GROUP in metric_dict:
-                try:
-                    node_id = metric_dict[SIGNAL_METRICS_GROUP]
-                    group_names.append(node_id)
+            try:
+                metric_name = metric_dict[SIGNAL_METRICS_METRIC_NAME]
+
+                group_names = []
+                if SIGNAL_METRICS_GROUP in metric_dict:
+                    root_group = metric_dict[SIGNAL_METRICS_GROUP]
+                    group_names.append(root_group)
 
                     if metric_dict[SIGNAL_METRICS_GROUP_DIMENSION].lower() != AGGREGATE:
```

Comment:
TODO: need to check whether GROUP_DIMENSION exists here. Currently, it throws KeyError, which is a bug.

Reply:
Done